.github/workflows/gcassert.yml

```diff
@@ -0,0 +1,25 @@
+name: gcassert
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.15
+
+    - name: Check out code
+      uses: actions/checkout@v1
+
+    - name: Run gcassert
+      run: go run github.com/jordanlewis/gcassert/cmd/gcassert
```

Comment:
I think I'd prefer to just this part of our existing tests so "go test ./..." catches it, rather than remembering to run some separate tool.

I'd fine importing `gcassert`'s guts in our testing code, though.

The standard library has a number of tests that check for this sort of stuff (running the compiler or `go tool nm`) and I always liked just having them part of a normal test.

Reply:
Seconded.

Reply:
Sure, looks like it'll be easy enough. Can do.

Reply:
IANAL, but here's the license header from the tool before I start importing its code:

```
// Copyright 2020 The Cockroach Authors.
//
// Use of this software is governed by the Business Source License
// included in the file licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with
// the Business Source License, use of this software will be governed
// by the Apache License, Version 2.0, included in the file
// licenses/APL.txt.
```

Any objections to that?

Reply:
/cc @jordanlewis

Reply:
Naah, let's not mix in non-open source code. I'll write it myself.

Reply:
Argh, this doesn't need to be BSL at all. I can change the license to something unrestrictive, I'll re-ping here when I do.

Reply:
I updated the license to MIT. I'm guessing you won't use it anyway, but on the off chance you want to, it's MIT licensed.

Reply:
@jordanlewis, thanks! I'll leave it to @mdlayher on whether he wants to update my https://github.com/inetaf/netaddr/commit/be7e645283661d9d64c0e6731e8df7820a9f3dc6 version with gcassert.